sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -375,6 +390,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
     val queries = tempQueries.map(_.trim).filter(_ != "").toSeq
       // Fix misplacement when comment is at the end of the query.
       .map(_.split("\n").filterNot(_.startsWith("--")).mkString("\n")).map(_.trim).filter(_ != "")
+      // Expand the queries if it's CTETest
+      .flatMap(query => expandCTEQuery(query, testCase))
```

Comment:
The problem of this approach is that we need to manually check the golden file and make sure the CTE query results are same with or without temp views.

Can we update `runQueries` instead? if the test is `CTETest`, we run the non-command query with temp view again and check if the result is the same as without temp view: https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L471-L478

Reply:
Thanks, that's a good point. I now do this comparison for those non-command queries with no exceptions. Since for queries with exception, the select from view will always return `view not found`.